File: src/1352-Product-of-the-Last-K-Numbers/1352.cpp
class ProductOfNumbers 
{
public:
    ProductOfNumbers() {}
    
    void add(int num) 
    {
        if (num == 0) pre = {1};
        else pre.push_back(pre.back() * num);
    }
    
    int getProduct(int k) 
    {
        if (pre.size() <= k) return 0;
        return pre.back() / pre[pre.size() - k - 1];
    }
    
private:
    vector<int> pre = {1};
};

File: src/1352-Product-of-the-Last-K-Numbers/1352.py
class ProductOfNumbers:
    def __init__(self):
        self.pre = [1]

    def add(self, num: int) -> None:
        if num == 0:
            self.pre = [1]
        else:
            self.pre.append(self.pre[-1] * num)

    def getProduct(self, k: int) -> int:
        if k >= len(self.pre): return 0
        return self.pre[-1] // self.pre[-k-1]